win32/src/socket_transport_win32.c

```diff
@@ -0,0 +1,720 @@
+            if (open_result != SM_EXEC_GRANTED)
+            {
+                LogError("sm_open_begin failed with %" PRI_MU_ENUM, MU_ENUM_VALUE(SM_RESULT, open_result));
+                result = MU_FAILURE;
+            }
+            else
+            {
+                // Codes_SOCKET_TRANSPORT_WIN32_09_060: [ socket_transport_listen shall call socket with the params AF_INET, SOCK_STREAM and IPPROTO_TCP. ]
+                socket_transport->socket = socket(AF_INET, SOCK_STREAM, IPPROTO_TCP);
+                if (socket_transport->socket == INVALID_SOCKET)
+                {
+                    LogLastError("Could not create socket");
+                    result = MU_FAILURE;
+                }
+                else
+                {
+                    u_long iMode = 1;
+                    struct sockaddr_in service;
+
+                    service.sin_family = AF_INET;
+                    service.sin_addr.s_addr = INADDR_ANY;
+                    service.sin_port = htons(port);
+
+                    // Codes_SOCKET_TRANSPORT_WIN32_09_061: [ socket_transport_listen shall bind to the socket by calling bind. ]
+                    if (bind(socket_transport->socket, (SOCKADDR*)&service, sizeof(service)) == SOCKET_ERROR)
+                    {
+                        LogLastError("Could not bind socket, port=%" PRIu16 "", port);
+                        result = MU_FAILURE;
+                    }
+
+                    // Codes_SOCKET_TRANSPORT_WIN32_09_065: [ sock_transport_listen shall set listening socket in non-blocking mode by calling ioctlsocket. ]
+                    else if (ioctlsocket(socket_transport->socket, FIONBIO, &iMode) != 0)
+                    {
+                        LogLastError("Could not set listening socket in non-blocking mode");
+                        result = MU_FAILURE;
+                    }
+                    else
+                    {
+                        // Codes_SOCKET_TRANSPORT_WIN32_09_062: [ socket_transport_listen shall start listening to incoming connection by calling listen. ]
+                        if (listen(socket_transport->socket, SOMAXCONN) == SOCKET_ERROR)
+                        {
+                            LogLastError("Could not start listening for connections");
+                            result = MU_FAILURE;
+                        }
+                        else
+                        {
+                            // Codes_SOCKET_TRANSPORT_WIN32_09_063: [ If successful socket_transport_listen shall call sm_open_end with true. ]
+                            sm_open_end(socket_transport->sm, true);
+                            result = 0;
+                            goto all_ok;
+                        }
+                    }
+                    // Codes_SOCKET_TRANSPORT_WIN32_09_066: [ socket_transport_listen shall call closesocket. ]
+                    closesocket(socket_transport->socket);
```

Comment:
>closesocket [](http://example.com/codeflow?start=20&length=11)

Need to put void in front of the closesocket:

(void)closesocket(socket_transport->socket);